### ISSUE TITLE
Qubes Manager shows that whonix-ws-dvm needs to be updated, but the update button is disabled

### ISSUE THREAD

**Issue by donob4n** [QubesOS/qubes-issues#3736]
### Qubes OS version: 

Qubes 4 testing

### Affected component(s): 

qubesmanager

---

### Steps to reproduce the behavior:

I don't know how to reproduce. I saw it since time ago.

### Expected behavior:

whonix-ws-dvm shouldn't need to be updated (at least with standard distribution procedure). Also, if Qube Manager considers it needs to be updated, should it let you click the update button? 

### Actual behavior:

You can see the update needed icon but you can't click the update button.

![update](https://user-images.githubusercontent.com/35405566/37827364-72169366-2e97-11e8-9f65-9bc4d6bbc64f.png)

### General notes:

I am not sure if Tor Browser should be installed in whonix-ws template or whonix-ws-dvm. In second case the update button could be useful for updating it.

---

**Comment by marmarta** [comment]
can you tell me the output of `qvm-prefs whonix-ws-dvm` and `qvm-features whonix-ws-dvm` (both run from dom0 console)? It would be really helpful in debugging this.

**Comment by donob4n** [comment]
Yes here is:

> [user@dom0 ~]$ qvm-prefs whonix-ws-dvm
> autostart             D  False
> backup_timestamp      U
> debug                 D  False
> default_dispvm        -  whonix-ws-dvm
> default_user          D  user
> gateway               D  
> gateway6              D  
> include_in_backups    D  True
> installed_by_rpm      D  False
> ip                    D  10.137.0.8
> ip6                   D  
> kernel                D  4.14.18-1
> kernelopts            D  nopat
> klass                 D  AppVM
> label                 -  red
> mac                   D  00:16:3E:5E:6C:00
> maxmem                D  4000
> memory                D  400
> name                  -  whonix-ws-dvm
> netvm                 -  sys-whonix
> provides_network      D  False
> qid                   -  8
> qrexec_timeout        D  60
> start_time            D  
> stubdom_mem           U
> stubdom_xid           D  -1
> template              -  whonix-ws
> template_for_dispvms  -  True
> updateable            D  False
> uuid                  -  434be95b-a5c2-4389-9bea-ce4094916d8c
> vcpus                 D  2
> virt_mode             D  pvh
> visible_gateway       D  10.137.0.7
> visible_gateway6      D  
> visible_ip            D  10.137.0.8
> visible_ip6           D  
> visible_netmask       D  255.255.255.255
> xid                   D  -1
> 
> [user@dom0 ~]$ qvm-features whonix-ws-dvm
> appmenus-dispvm    1
> updates-available  1
>

**Comment by marmarta** [comment]
@marmarek , check this out - a VM that got both updateable=False and updates-available=1 . I think there's a backend problem with assigning the updates-available feature.

**Comment by Wikinaut** [comment]
I have the same problem. 

But in addition, when I want to ```delete``` the disposable vm in order to clean up the mess, then I see an 
```
ERROR: Domain is in use: '(dvm-name)'; details in system.log
```

**Comment by DanMcLaughlin** [comment]
Yeah I'm seeing the same issue in 4.0 release, both for fedora-26-dvm and whonix-wx-dvm. I update both parent templates and the dvm's show an update icon, but can't hit the update. On starting the dvm's they don't automatically regenerate the image file as they did in version 3

**Comment by donob4n** [comment]
Maybe this is not the root of the problem, but this is fixed in https://github.com/QubesOS/qubes-manager/pull/91 since it only checks 'updates-available' for 'TemplateVM' domains.

**Comment by DanMcLaughlin** [comment]
@andrewdavidwong Could you post a link or a short note on how to manually regenerate the image files? I've done a bunch of searching but for Qubes 4 aren't finding the secret sauce unfortunately. Thanks - I'd like to at least get them updated ASAP.

**Comment by donob4n** [comment]
@DanMcLaughlin I think this is only a qube manager notification error (due some backend problem) but updating your fedora template you have already your dvm updated too.

**Comment by DanMcLaughlin** [comment]
@donob4n Ah, no wonder I couldn't find anything, makes sense. Thanks -

**Comment by donob4n** [comment]
You can use AppVM's as DispVM base for do customs configurations. See https://www.qubes-os.org/doc/dispvm-customization/

**Comment by andrewdavidwong** [comment]
@DanMcLaughlin: @donob4n provided the correct link to describe the new behavior in Qubes 4.0.

**Comment by donob4n** [comment]
Should be fixed with https://github.com/QubesOS/qubes-manager/pull/91

**Comment by Zig-03** [comment]
I have the same issue.

**Comment by donob4n** [comment]
Do you mean whonix-ws-dvm and whonix-ws?

Did you try restarting Qube Manager? I think there is some problem with sync but in any cash whonix-ws-dvm should not get update icon. Do you have refresh button?

**Comment by exidez** [comment]
I have the same issue with Zig-03 and others. Still unresolved. I am running a fresh install of Qubes R4.0 and noticed this after updating all the template VM's. Everything is fully updated including dom0. Update symbol on whonix-ws-dvm will not go away.

![screenshot_2018-07-14_21-38-47](https://user-images.githubusercontent.com/24729621/42724075-2898a9c4-87af-11e8-9817-dc90c1940da4.png)

**Comment by exidez** [comment]
I am having the same issue as Zig-03 and other. I have a fresh install of Qubes R4.0 and have updated all the template VM's and dom0. I noticed this after the updates and whonix-dw-dvm is still showing that it needs an update.

![screenshot_2018-07-14_21-38-47](https://user-images.githubusercontent.com/24729621/42724140-21e8e8b8-87b0-11e8-9238-cb1523a6197a.png)

**Comment by techgeeknz** [comment]
I noticed this bug too, but never got around to reporting it.  In my use-case, I have created extra DVMs for making quick tests of website changes in a clean web browser.

Having just now enabled the dom0-testing repository and updating (to test a different bug fix), I can confirm that this problem is fixed—Qube Manager was showing updates for all my DVMs, I restarted it, and now the "updates-available" flags are correctly attributed to the DVM templates.

Since I use DVMs frequently, I'll clone of one of my templates, make a DVM from it, and watch to see if the problem recurs next time an update is available.
